dockerfiles/platform-build-incremental-container/Dockerfile

```diff
@@ -22,20 +22,50 @@ ARG CR_REPO=docker.io
 ARG PLATFORM_PRBASE_VER
 FROM ${CR_REPO}/${CR_USER}/smoketest-platform-build:${PLATFORM_PRBASE_VER}
 
+ENV DEBIAN_FRONTEND=noninteractive
+ENV VCPKG_BINARY_SOURCES="clear;nuget,GitHub,read"
+ENV VCPKG_NUGET_REPOSITORY=https://github.com/hpcc-systems/vcpkg
+
+RUN apt-get update
+RUN apt-get install -y dirmngr gnupg apt-transport-https ca-certificates software-properties-common
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+RUN apt-add-repository 'deb https://download.mono-project.com/repo/ubuntu stable-focal main'
+RUN apt install -y mono-complete 
+
 USER hpcc
 
 WORKDIR /hpcc-dev/HPCC-Platform
 ARG GITHUB_REPO
 RUN git remote add upstream https://github.com/${GITHUB_REPO}
 ARG GITHUB_PRREF
 RUN echo GITHUB_PRREF=${GITHUB_PRREF}
+ARG GITHUB_OWNER
+RUN echo GITHUB_OWNER=${GITHUB_OWNER}
+ARG GITHUB_TOKEN
+RUN echo GITHUB_TOKEN=${GITHUB_TOKEN}
 RUN echo git fetch upstream +${GITHUB_PRREF}
 RUN git fetch upstream +${GITHUB_PRREF}
 RUN git checkout FETCH_HEAD
 RUN git submodule update --init --recursive
 
+WORKDIR /hpcc-dev/HPCC-Platform/vcpkg
+RUN ./bootstrap-vcpkg.sh
+
+RUN mono `./vcpkg fetch nuget | tail -n 1` \
+    sources add \
+    -name "GitHub" \
+    -source "https://nuget.pkg.github.com/hpcc-systems/index.json" \
+    -storepasswordincleartext \
+    -username "${GITHUB_OWNER}" \
+    -password "${GITHUB_TOKEN}"
+
 WORKDIR /hpcc-dev/build
 
+ARG BUILD_TYPE=RelWithDebInfo
+ARG USE_CPPUNIT=1
+RUN rm CMakeCache.txt
+RUN cmake /hpcc-dev/HPCC-Platform -Wno-dev -DCMAKE_TOOLCHAIN_FILE=/hpcc-dev/HPCC-Platform/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_OVERLAY_PORTS=/hpcc-dev/HPCC-Platform/vcpkg-overlays -DCONTAINERIZED=1 -DINCLUDE_PLUGINS=1 -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DUSE_PYTHON2=0 -DSUPPRESS_SPARK=1 -DUSE_CPPUNIT=${USE_CPPUNIT}
+
```

Comment:
@jakesmith - I think this was needed as the currently cached base does include the VCPKG settings?

Reply:
I have updated the cache id to:  "platform_build::smoketest-platform-build-vcpkg" so they won't clash while we are migrating - which now means the above is not needed (I will remove and test)?